storage/src/adb/any/variable/sync.rs

```diff
@@ -34,17 +34,22 @@ use tracing::debug;
 /// - `items_per_section`: number of items per section
 ///
 /// # Returns
-/// A journal whose sections satisfy:
+/// (journal, size) where:
 /// - No section index < `lower_bound / items_per_section` exists.
 /// - No section index > `upper_bound / items_per_section` exists.
-/// - The last retained section is truncated so that its last item’s location is `<= upper_bound`.
+/// - No item with location > `upper_bound` exists.
+/// - `size` is the next location that should be appended to by the sync engine.
+///
+/// # Errors
+/// Returns [Error::InvalidSyncRange] if lower_bound > upper_bound.
+/// Returns [Error::UnexpectedData] if existing data extends beyond `upper_bound`.
 pub(crate) async fn init_journal<E: Storage + Metrics, V: Codec>(
```

Comment:
As we think about actually integrating this, we'll need to store the last sync target we fed to journal somewhere (to reinitialize).

I suppose commenting on this nuance is best handled in some place where we actually do this (and the conditions here are sufficient to imply this).